File: ruby/leap/leap.rb
# assignment 10: leap
# author: Jason Morse
# date: June 6, 2015; updated: June 13, 2015
#

class Year

  def self.leap?(year)
    (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)
  end

end

File: javascript/leap/leap.js
// file: leap.js (assignment 2)
// author: Jason Morse
// date: 12/24/2015
//

// constructor to create 'Year' objects
var Year = function(year) {
	this.year = year;
};

// define 'isLeap' in the prototype so all 'Year' objects inherit it
Year.prototype.isLeap = function() {
	return ((this.year % 4 == 0) && (this.year % 100 != 0)) || (this.year % 400 == 0);
}

module.exports = Year;